### DISCUSSION TITLE
流水线中使用远程缓存每次启动流水线都会创建一个新的session 导致缓存完全不生效

### DISCUSSION THREAD

**Discussion by ZYC97104** [xmake-io/xmake#4524] [Q&A]
xmake version 2.8.5
流水线为docker编译模式 所以每次启动流水线都是原始的编译环境
在构建脚本内添加远程缓存地址并连接
![image](https://github.com/xmake-io/xmake/assets/54052023/8f02e455-6877-4a8b-933c-2964c2eb8f8b)

每次构建都会生成新的缓存
![Uploading image.png…]()

请问是使用姿势有问题吗 在流水线中该怎么使用远程缓存？

**Comment by waruqi** [comment]
首次连接成功，会把当前连接的 session id 缓存到 .xmake/service/remote_cache/status.txt 

之后的每次重连，都会复用本地缓存的 session id，确保session id 相同。。

但是如果你 docker 里面每次都是开新工程连接，或者工程的缓存都清了，那么每次肯定会是新的一个 session.

目前缓存是跟工程强绑定的，不是系统全局的

**Comment by waruqi** [comment]
我加了个 `--session=xxx` 参数，可以自己指定 session name 生成 session id，你可以 connect 的时候，指定相同的 session ，这样就能共用了。。

```
xmake update -s dev
xmake service --connect --ccache --session=myproj
```

**Comment by waruqi** [comment]
现有版本远程 缓存有点问题，每次都会重新  build + push。会很慢，建议使用 dev 版本 

https://github.com/xmake-io/xmake/issues/4592
